front/app/components/EsriMap/index.tsx

```diff
@@ -98,144 +80,122 @@ const EsriMap = ({
-            breakpoint: false,
-          },
-        },
-      });
+    if (!mapRefAvailable) return;
 
-      setMap(newMap);
-      setWebMap(webMap || null);
-      setMapView(mapView);
+    let map: Map | WebMap;
 
-      return () => {
-        mapView.destroy();
-      };
+    if (webMapId) {
+      map = new WebMap({ portalItem: { id: webMapId } });
+    } else {
+      map = new Map();
+      map.basemap = new Basemap({
+        baseLayers: [getDefaultBasemap(globalMapSettings.tile_provider)],
+      });
     }
 
-    return;
-  }, [webMapId]);
+    setMap(map);
+
+    const mapView = new MapView({
+      container: mapRef.current as HTMLDivElement, // Reference to DOM node that will contain the view
+      map,
+      popupEnabled: false,
+      popup: {
+        dockEnabled: false,
+        dockOptions: {
+          breakpoint: false,
+        },
+      },
+    });
+
+    setMapView(mapView);
+    setUpdateMapViewConfig(true);
+
+    return () => {
+      mapView.destroy();
+    };
+  }, [mapRefAvailable, webMapId, globalMapSettings.tile_provider]);
 
   // Load initial map configuration data that was passed in.
-  // Note: This data is static and will not change.
+  // This will run on initial render and whenever the
+  // webMapId changes
   useEffect(() => {
-    if (!initialValuesLoaded.current && mapView && map) {
-      // Set map center
-      setMapCenter(mapView, initialData, globalMapSettings);
-
-      // Set the basemap & initial extent
-      map.basemap = new Basemap({
-        baseLayers: [getDefaultBasemap(globalMapSettings.tile_provider)],
-      });
-      mapView.zoom = initialData?.zoom || globalMapSettings.zoom_level || 18;
-      mapView.constraints = {
-        maxZoom: initialData?.maxZoom || 22,
-        minZoom: 5,
-      };
-
-      // Change location of zoom widget if specified
-      if (initialData?.zoomWidgetLocation === 'right') {
-        const zoom = mapView.ui.find('zoom');
-        mapView.ui.add(zoom, 'top-right');
-      }
-
-      // Add fullscreen widget if set
-      if (initialData?.showFullscreenOption) {
-        const fullscreen = new Fullscreen({
-          view: mapView,
-        });
-        mapView.ui.add(fullscreen, 'top-right');
-      }
-
-      // Add map legend if set
-      if (initialData?.showLegend) {
-        addMapLegend(mapView, isMobileOrSmaller);
-      }
-
-      // Show layer visibility controls if set
-      if (initialData?.showLayerVisibilityControl) {
-        showLayerVisibilityControls(mapView);
-      }
-
-      // Add any ui elements that were passed in
-      if (initialData?.uiElements && mapView) {
-        initialData?.uiElements.forEach((uiElement) => {
-          mapView.ui.add(uiElement.element, uiElement.position);
-        });
-      }
+    if (!mapView || !updateMapViewConfig) return;
 
-      // Run onInit function if it was provided
-      if (initialData?.onInit) {
-        initialData.onInit(mapView);
-      }
+    configureMapView(
```

Comment:
We apply all the settings once when `setUpdateMapViewConfig(false)` is called (which only happens in the `useEffect` above)